src/main/java/com/eppo/sdk/EppoClient.java

```diff
@@ -121,16 +122,136 @@ public Optional<String> getAssignment(
      * @param experimentKey
      * @return
      */
-    public Optional<String> getAssignment(String subjectKey, String experimentKey) {
+    private Optional<EppoValue> getAssignment(String subjectKey, String experimentKey) {
         return this.getAssignment(subjectKey, experimentKey, new SubjectAttributes());
     }
 
+    /**
+     * This function will return typed assignment value
+     * @param subjectKey
+     * @param experimentKey
+     * @param type
+     * @param subjectAttributes
+     * @return
+     */
+    private Optional<?> getTypedAssignment(String subjectKey, String experimentKey, EppoValueType type, SubjectAttributes subjectAttributes) {
+        Optional<EppoValue> value = this.getAssignment(subjectKey, experimentKey, subjectAttributes);
+        if (value.isEmpty()) {
+            return Optional.empty();
+        }
+
+        switch (type) {
+            case BOOLEAN:
+                return Optional.of(value.get().boolValue());
+            case NUMBER:
+                return Optional.of(value.get().longValue());
+            default:
+                return Optional.of(value.get().stringValue());
+        }
+    }
+
+    /**
+     * This function will return typed assignment value without subject attribute
+     * @param subjectKey
+     * @param experimentKey
+     * @param type
+     * @return
+     */
+    private Optional<?> getTypedAssignment(String subjectKey, String experimentKey, EppoValueType type) {
+        return this.getTypedAssignment(subjectKey, experimentKey, type, new SubjectAttributes());
+    }
+
+    /**
+     * This function will return string assignment value
+     * @param subjectKey
+     * @param experimentKey
+     * @param subjectAttributes
+     * @return
+     */
+    public Optional<String> getStringAssignment(String subjectKey, String experimentKey, SubjectAttributes subjectAttributes) {
+       return (Optional<String>) this.getTypedAssignment(subjectKey, experimentKey, EppoValueType.STRING, subjectAttributes);
+    }
+
+    /**
+     * This function will return string assignment value without passing subjectAttributes
+     * @param subjectKey
+     * @param experimentKey
+     * @return
+     */
+    public Optional<String> getStringAssignment(String subjectKey, String experimentKey) {
+        return this.getStringAssignment(subjectKey, experimentKey, new SubjectAttributes());
+    }
+
+    /**
+     * This function will return boolean assignment value
+     * @param subjectKey
+     * @param experimentKey
+     * @param subjectAttributes
+     * @return
+     */
+    public Optional<Boolean> getBooleanAssignment(String subjectKey, String experimentKey, SubjectAttributes subjectAttributes) {
```

Comment:
I think the non-string `get<Type>Assignments()` functions don't have unit tests. While the code is simple/sound, I'd consider whipping up a few more test cases to protect future us.